### PR TITLE
MorphTo Refactor

### PR DESCRIPTION
**Pull request by ryandialpad**
#### Type of PR:

- [ ] Bugfix
- [x] Feature
- [x] Refactor
- [ ] Code style update
- [ ] Build-related changes
- [ ] Test
- [ ] Documentation
- [ ] Other, please describe:

#### Breaking changes:

- [x] No
- [ ] Yes

### Details

Implementing changes discussed in the following draft PR: https://github.com/vuex-orm/vuex-orm-next/pull/107

We are going to want to squash and merge this since I am a frequent committer 🙃

### CODE REVIEW

File: src/model/attributes/relations/MorphTo.ts [hunk 1]
```diff
@@ -101,40 +102,36 @@ export class MorphTo extends Relation {
...
+      // Otherwise, set model relations using the dictionary
+      models.forEach((model) => {
+        const type = model[this.morphType]
+        const id = model[this.morphId]
+
+        const related = dictionary[type]?.[id] ?? null
+
+        model.$setRelation(relation, related)
+      })
+    }
```
Comment (cuebit) [inline src/model/attributes/relations/MorphTo.ts:134]: I don't think we need this if/else statement. Seems redundant as the relation will be null if it's not indexed in the dictionary. Let's keep it simple and use one loop branch here. What do you think?

**Review by cuebit — CHANGES_REQUESTED** [review]
Just one comment to review 🙏 . Generally, @kiaking will avoid if/else conditions where possible (you may notice there is only one place this exists!) 😄

File: src/schema/Schema.ts [hunk 2]
```diff
@@ -116,7 +116,7 @@ export class Schema {
       // missing foreign keys to the record first.
       if (key !== null) {
         // Using optional chaining in the event of an inverse polymorphic
-        // relation that won't have a defined relation.
+        // relation which won't have a defined relation.
         ;(parent.$fields()[key] as Relation)?.attach(parentRecord, record)
```
Comment (cuebit) [inline src/schema/Schema.ts:120]: Also, we can definitely remove the optional chain here as per [this comment](https://github.com/vuex-orm/vuex-orm-next/pull/107#discussion_r753888042).

**Review by cuebit — CHANGES_REQUESTED** [review]

File: src/model/attributes/relations/MorphTo.ts (hunk 1 above)
Comment (ryandialpad) [reply src/model/attributes/relations/MorphTo.ts:134]: I wasn't 100% sure if we wanted to favour the simpler approach and opt'd to attempt to optimize a bit. Personally, I prefer the simpler approach 👍

Comment (cuebit) [reply src/model/attributes/relations/MorphTo.ts]: Awesome 🎉

**Review by cuebit — APPROVED** [review]
Amazing job 👏 this looks and works like a charm 👍

**Comment by ryandialpad** [comment]
> Amazing job 👏 this looks and works like a charm 👍

Thanks! Couldn't have done it without @kiaking and you 😄

**Comment by cuebit** [comment]
Merged into #106. @kiaking can review and manage from there 👍

Thanks for all your hard work @ryandialpad 🙌

**Pull request merged**